loadbalancer/algorithm.go

```diff
@@ -40,49 +41,149 @@ var (
 	defaultAlgorithm = newRoundRobin
 )
 
-func newRoundRobin(endpoints []string) routing.LBAlgorithm {
-	i := time.Now().UnixNano()
-	rand.Seed(i)
-	return &roundRobin{
-		index: rand.Intn(len(endpoints)),
+func fadeInState(now time.Time, duration time.Duration, detected time.Time) (time.Duration, bool) {
+	rel := now.Sub(detected)
```

Comment:
Can this be negative?

Reply:
In practice, not really, because we capture the detected timestamp, in-process. OTOH, the structure defining the input field is public, so it is a good point to validate it. Thx, will add a check.